docs/management_api.yml

```diff
@@ -0,0 +1,127 @@
+swagger: '2.0'
+info:
+  version: '0.1.0'
+  title: Device authentication
+  description: |
+      An API for device authentication handling. Not exposed via the API Gateway - intended for internal use only.
+
+basePath: '/api/0.1.0/'
+host: 'mender-device-auth:8080'
+schemes:
+  - http
+
+paths:
+  /devices:
+    get:
+      summary: Get a list of tenant's devices.
+      description: |
+        Provides a list of tenant's devices, with optional device status filter.
+      parameters:
+        - name: page
+          in: query
+          description: Results page number
+          required: false
+          type: number
+          format: integer
+          default: 1
+        - name: per_page
+          in: query
+          description: Number of results per page
+          required: false
+          type: number
+          format: integer
+          default: 20
+      responses:
+        200:
+          description: An array of devices.
+          schema:
+            type: array
+            items:
+                $ref: '#/definitions/Device'
+          headers:
+            Link:
+              type: string
+              description: Standard header, we support 'first', 'next', and 'prev'.
+        400:
+          description: Missing/malformed request params.
+          schema:
+            $ref: '#/definitions/Error'
+        404:
+          description: Tenant not found.
+          schema:
+            $ref: '#/definitions/Error'
+        500:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+  /devices/{id}:
+    get:
+      summary: Get a particular device.
+      parameters:
+        - name: id
+          in: path
+          description: Device identifier
+          required: true
+          type: string
+      responses:
+        200:
+          description: Device found.
+          schema:
+            $ref: '#/definitions/Device'
+        404:
+          description: Device not found.
+          schema:
+            $ref: '#/definitions/Error'
+        500:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+definitions:
+  Device:
+    type: object
+    properties:
+      id:
+        type: string
+        description: Mender assigned Device ID.
+      tenant_token:
```

Comment:
@mchalski should tenant_token be here? All devices would have same within request.